### PR TITLE
OCPBUGS-33882: Streamline bootstrap crds

### PR DESCRIPTION
**Pull request by JoelSpeed**
Now that we are adding additional CRDs to the API image, they are presently being added via the bootstrap rendering to the bootstrap node installation of manifests.
This does not presently account for the fact that some CRDs may not want to be installed (eg capability disabled).

To mitigate this, this PR is reducing the number of PRs installed at the bootstrap time by restricting the CRDs rendered at the installation time, and deferring all other CRDs to be installed by the CVO, which is already aware of capabilities.

This will also help reduce the bootstrap node responsibilities and footprint.

After discussion with @deads2k, keeping all the existing config v1 CRDs plus the minimal set outside of that group that are needed, so, the ones being dropped from bootstrap are:
* RangeAllocation
* Config (operator v1)
* ImageContentSourcePolicies (operator v1)
* CPMS (was never meant to be)

### CODE REVIEW

File: authorization/v1/types.go [hunk 1]
```diff
@@ -537,6 +537,7 @@ type ClusterRoleList struct {
 // +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/470
 // +openshift:file-pattern=cvoRunLevel=0000_03,operatorName=config-operator,operatorOrdering=01
 // +openshift:compatibility-gen:level=1
+// +kubebuilder:metadata:annotations=release.openshift.io/bootstrap-required=true
 type RoleBindingRestriction struct {
```
Comment (deads2k) [inline authorization/v1/types.go:541]: oh neat, I had forgotten about this one.  I wonder how many people use this feature.

**Comment by JoelSpeed** [comment]
Spoke to OTA in [slack](https://redhat-internal.slack.com/archives/CEGKQ43CP/p1716308792259719), no objections to the current proposed naming.

Raised a [bug](https://issues.redhat.com/browse/OCPBUGS-34059) for the api-server team to fix `cluster-bootstrap`.

/retest
/payload nightly blocking

File: payload-command/render/renderassets/assets.go [hunk 2]
```diff
@@ -177,6 +177,31 @@ func ClusterProfile(clusterProfile string) FileContentsPredicate {
...
+// BootstrapRequiredCRD returns a predicate for LoadFilesRecursively that filters manifests
+// based on whether they are marked as being required for bootstrap or not.
+func BootstrapRequiredCRD() FileContentsPredicate {
+	return func(manifest []byte) (bool, error) {
+		uncastObj, _, err := codecs.UniversalDecoder().Decode(manifest, nil, &unstructured.Unstructured{})
+		if err != nil {
+			panic(fmt.Errorf("unable to decode: %w", err))
+		}
+
+		isBootstrapCRD := uncastObj.(*unstructured.Unstructured).GetAnnotations()["release.openshift.io/bootstrap-required"]
```
Comment (deads2k) [inline payload-command/render/renderassets/assets.go:189]: don't we only provide manifests in our repo that are for bootstrapping?  Why do we need to filter here versus at the bash script that does the copy?

**Pull request merged**